sdk/core/System.ServiceModel.Rest/src/Temporary/ModelSerializationExtensions.cs

```diff
@@ -0,0 +1,347 @@
+                case IUtf8JsonWriteable serializable:
+                    serializable.Write(writer);
+                    break;
+                case byte[] bytes:
+                    writer.WriteBase64StringValue(bytes);
+                    break;
+                case BinaryData bytes:
+                    writer.WriteBase64StringValue(bytes);
+                    break;
+                case System.Text.Json.JsonElement json:
+                    json.WriteTo(writer);
+                    break;
+                case int i:
+                    writer.WriteNumberValue(i);
+                    break;
+                case decimal d:
+                    writer.WriteNumberValue(d);
+                    break;
+                case double d:
+                    if (double.IsNaN(d))
+                    {
+                        writer.WriteStringValue("NaN");
+                    }
+                    else
+                    {
+                        writer.WriteNumberValue(d);
+                    }
+                    break;
+                case float f:
+                    writer.WriteNumberValue(f);
+                    break;
+                case long l:
+                    writer.WriteNumberValue(l);
+                    break;
+                case string s:
+                    writer.WriteStringValue(s);
+                    break;
+                case bool b:
+                    writer.WriteBooleanValue(b);
+                    break;
+                case Guid g:
+                    writer.WriteStringValue(g);
+                    break;
+                case DateTimeOffset dateTimeOffset:
+                    writer.WriteStringValue(dateTimeOffset, "O");
+                    break;
+                case DateTime dateTime:
+                    writer.WriteStringValue(dateTime, "O");
+                    break;
+                case IEnumerable<KeyValuePair<string, object>> enumerable:
+                    writer.WriteStartObject();
+                    foreach (KeyValuePair<string, object> pair in enumerable)
+                    {
+                        writer.WritePropertyName(pair.Key);
+                        writer.WriteObjectValue(pair.Value);
+                    }
+                    writer.WriteEndObject();
+                    break;
+                case IEnumerable<object> objectEnumerable:
+                    writer.WriteStartArray();
+                    foreach (object item in objectEnumerable)
+                    {
+                        writer.WriteObjectValue(item);
+                    }
+                    writer.WriteEndArray();
+                    break;
+                case TimeSpan timeSpan:
+                    writer.WriteStringValue(timeSpan, "P");
+                    break;
+
+                default:
+                    throw new NotSupportedException("Not supported type " + value.GetType());
+            }
+        }
+        private class TypeFormatters
```

Comment:
This is entirely duplicated, and we need to address that.  It'd be great to get this logic into a public type, but not sure yet what the API surface looks like.